frontend/src/components/ShootDetails/ShootKubeconfig.vue

```diff
@@ -112,23 +136,85 @@ export default {
     isGardenloginType () {
       return this.type === 'gardenlogin'
     },
+    isStaticKubeconfigLoginType () {
+      return this.type === 'token'
+    },
+    isAdminKubeconfigLoginType () {
+      return this.type === 'adminkubeconfig'
+    },
+    possibleAdminKubeconfigExpirationSettings () {
+      const defaultExpirations = ['30m', '1h', '3h', '6h', '12h', '1d', '3d', '7d']
+      let filteredExpirations = defaultExpirations
+
+      if (this.cfg.adminKubeconfig.maxExpirationSeconds) {
+        filteredExpirations = []
+        defaultExpirations.forEach((val) => {
+          if (this.adminKubeConfigExpirationInSeconds(val) <= this.cfg.adminKubeconfig.maxExpirationSeconds) {
+            filteredExpirations.push(val)
+          }
+        })
+      }
+
+      return filteredExpirations
+    },
     getQualifiedName () {
       const prefix = this.isGardenloginType ? 'kubeconfig-gardenlogin' : 'kubeconfig'
       return `${prefix}--${this.shootProjectName}--${this.shootName}.yaml`
     }
   },
   methods: {
-    toggleKubeconfig () {
+    async createCachedAdminKubeconfig () {
+      if (!this.adminKubeConfig) {
+        try {
+          const resp = await createAdminKubeconfig({
+            namespace: this.shootNamespace,
+            name: this.shootName,
+            data: {
+              expirationSeconds: this.adminKubeConfigExpirationInSeconds(this.adminKubeConfigExpiration)
+            }
+          })
+
+          this.adminKubeConfig = Buffer.from(resp.data.status.kubeconfig, 'base64').toString('utf8')
+        } catch (err) {
+          const errorMessage = 'Could not request admin kubeconfig'
+          const errorDetails = errorDetailsFromError(err)
+          console.error(errorMessage, errorDetails.errorCode, errorDetails.detailedMessage, err)
+        }
+      }
+    },
+    async toggleKubeconfig () {
+      if (this.isAdminKubeconfigLoginType) {
+        await this.createCachedAdminKubeconfig()
```

Comment:
- button should display as loading/progressing when pressing the view button while the kubeconfig is being fetched
- Only when visible it is required to fetch the kubeconfig
- Do not cache the kubeconfig. Just fetch a new one. E.g. User could leave the page open for some time and click the download button again but would get possibly an expired kubeconfig

Reply:
How would I mark a button as loading / progressing?
I implemented the other points.

Reply:
see v-progress-circular. However I noticed that we do not do the same for the kubeconfigs that can be downloaded on the member page. If we add it then it should be added to the member page as well (but not in this PR). I resolve this conversation..